src/nimBuild.nim

```diff
@@ -140,13 +141,28 @@ proc parseErrors(lines: seq[cstring]): seq[CheckResult] =
       if messageText.len < 1024:
         messageText &= nodeOs.eol & line
     else:
-      let
+      var
         file = match[2]
         lineStr = match[3]
         charStr = match[5]
+      let
         severity = match[7]
         msg = match[8]
 
+      # file may be undefined when there's an error in code
+      # created with parseExpr/parseStmt
+      # as a workaround we duplicate the last location in the stacktrace
+      # There always has to be atleast the location where the macro
+      # called which contains the faulty parseExpr/parseStmt
+      if file.toJs() == jsUndefined:
```

Comment:
Instead how about lifting the predicate into the let, you can use the predicate and your unpacking to conditionally assign the three variable bits. That way no vars, decision and mutation are kept to a minimum.